pkg/volume/csi/csi_plugin.go

```diff
@@ -785,13 +798,61 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	return false, nil
 }
 
-// getDriverMode returns the driver mode for the specified spec: {persistent|ephemeral}.
+// supportsVolumeMode looks up CSIDriver object associated with the
+// driver name to determine if the driver supports the given volume
+// mode. An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode csiVolumeMode) error {
+	driverMode, isDefault, err := p.lookupDriverMode(driver)
+	switch {
+	case err != nil:
+		return err
+	case volumeMode == persistentVolumeMode && (driverMode == storage.PersistentDriverMode || driverMode == storage.CombinedDriverMode) ||
+		volumeMode == ephemeralVolumeMode && (driverMode == storage.EphemeralDriverMode || driverMode == storage.CombinedDriverMode):
+		return nil
+	case isDefault:
+		// Using the same error message might be confusing because 'which only supports "persistent"' would
+		// be wrong for a driver that merely doesn't specify that it supports ephemeral.
+		return fmt.Errorf("volume mode %q not supported by driver %s, no CSIDriver object or no Mode specified", volumeMode, driver)
+	default:
+		return fmt.Errorf("volume mode %q not supported by driver %s which only supports %q", volumeMode, driver, driverMode)
+	}
+}
+
+// lookupDriverMode looks up CSIDriver object associated with the driver name and
+// returns what it finds there or the "persistent" default.
```

Comment:
Please clarify in comment what the `bool`  return param is for is for.

Reply:
Will do. It tells the caller whether the returned string was explicitly set or is the default value. That's relevant when it comes to producing a good error message.

Would it be better to return the empty string for "not set" and drop the bool?

Reply:
> Would it be better to return the empty string for "not set" and drop the bool?

I changed it to that. @vladimirvivien: okay?